pkg/controller/stackconfigpolicy/controller.go

```diff
@@ -414,98 +541,159 @@ func (r *ReconcileStackConfigPolicy) updateStatus(ctx context.Context, scp polic
+	softOwner types.NamespacedName,
+	configuredESResources esMap,
+	configuredKibanaResources kbMap,
+	resourceType policyv1alpha1.ResourceType,
+) error {
+	err := resetOrphanSoftOwnedFileSettingSecrets(ctx, c, softOwner, configuredESResources, resourceType)
 	if err != nil {
 		return err
 	}
-	return deleteOrphanSoftOwnedSecrets(ctx, c, softOwner, configuredEs)
+	return deleteOrphanSoftOwnedSecrets(ctx, c, softOwner, configuredESResources, configuredKibanaResources, resourceType)
 }
 
-// resetOrphanSoftOwnedSecrets resets secrets for the Elasticsearch clusters that are no longer configured
+// resetOrphanSoftOwnedFileSettingSecrets resets secrets for the Elasticsearch clusters that are no longer configured
 // by a given StackConfigPolicy.
 // An optional list of Elasticsearch currently configured by the policy can be provided to filter secrets not to be modified. Without list,
 // all secrets soft owned by the policy are reset.
-func resetOrphanSoftOwnedSecrets(ctx context.Context, c k8s.Client, softOwner types.NamespacedName, configuredEs esMap) error {
+func resetOrphanSoftOwnedFileSettingSecrets(
+	ctx context.Context,
+	c k8s.Client,
+	softOwner types.NamespacedName,
+	configuredESResources esMap,
+	resourceType policyv1alpha1.ResourceType,
+) error {
 	log := ulog.FromContext(ctx)
 	var secrets corev1.SecretList
+	matchLabels := client.MatchingLabels{
+		reconciler.SoftOwnerNamespaceLabel:              softOwner.Namespace,
+		reconciler.SoftOwnerNameLabel:                   softOwner.Name,
+		reconciler.SoftOwnerKindLabel:                   policyv1alpha1.Kind,
+		commonlabels.StackConfigPolicyOnDeleteLabelName: commonlabels.OrphanSecretResetOnPolicyDelete,
+	}
+
+	if resourceType != "" {
+		matchLabels[commonv1.TypeLabelName] = string(resourceType)
+	}
+
 	if err := c.List(ctx,
 		&secrets,
 		// search in all namespaces
 		// restrict to secrets on which we set the soft owner labels
-		client.MatchingLabels{
-			reconciler.SoftOwnerNamespaceLabel:              softOwner.Namespace,
-			reconciler.SoftOwnerNameLabel:                   softOwner.Name,
-			reconciler.SoftOwnerKindLabel:                   policyv1alpha1.Kind,
-			commonlabels.StackConfigPolicyOnDeleteLabelName: commonlabels.OrphanSecretResetOnPolicyDelete,
-		},
+		matchLabels,
 	); err != nil {
 		return err
 	}
 	for i := range secrets.Items {
 		s := secrets.Items[i]
+		configuredApplicationType := s.Labels[commonv1.TypeLabelName]
+		switch configuredApplicationType {
+		case eslabel.Type:
+			namespacedName := types.NamespacedName{
+				Namespace: s.Namespace,
+				Name:      s.Labels[eslabel.ClusterNameLabelName],
+			}
+			if _, exists := configuredESResources[namespacedName]; exists {
+				continue
+			}
 
-		esNsn := types.NamespacedName{
-			Namespace: s.Namespace,
-			Name:      s.Labels[eslabel.ClusterNameLabelName],
-		}
-		_, exist := configuredEs[esNsn]
-		if exist {
-			continue
-		}
+			log.V(1).Info("Reconcile empty file settings Secret for Elasticsearch",
```

Comment:
I don't think it comes from your PR but we should take care of extra keys like `namespace` which are already included in the logger fetched from `ctx`:

```json
{"service.version": "0.0.0-SNAPSHOT+00000000", "iteration": "1", "namespace": "elastic-system", "policy_name": "test-stack-config-policy", "namespace": "es-ns", "es_name": "sample", "owner_namespace": "elastic-system", "owner_name": "test-stack-config-policy"}
```

`namespace` key is set twice which means that the generated JSON document is not valid.

Reply:
thanks let me make sure that is removed.